operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -1,120 +1,393 @@
-   same as the initial invocation of `iuf run` as all steps in this example use the same IUF activity, media, and variables.
+## Image preparation
+
+This section creates CFS configurations and bootable images that will be used by later steps in the workflow.
+
+Before proceeding, ensure any site customizations to product content stored in VCS have been made per [Perform manual product configuration operations](#perform-manual-product-configuration-operations) to ensure CFS configurations
+and images are created with the correct content and configuration values.
+
+### Execute the IUF `update-cfs-config` and `prepare-images` stages
+
+**`NOTE`** Additional arguments are available to control the behavior of the `update-cfs-config` and `prepare-images` stages, for example `-bc`, `-bm`, and `-rv`. See the [`update-cfs-config` stage
+documentation](../stages/update_cfs_config.md) and the [`prepare-images` stage documentation](../stages/prepare_images.md) for details and adjust the examples below if necessary.
+
+1. Refer to the "Install and Upgrade Framework" section of each individual product's installation documentation to determine if any special actions need to be performed outside of IUF for the `update-cfs-config` or
+`prepare-images` stages.
+
+1. Invoke `iuf run` with `-r` to execute the [`update-cfs-config`](../stages/update_cfs_config.md) and [`prepare-images`](../stages/prepare_images.md) stages. Use site variables from the `site_vars.yaml` file found in
+`${ADMIN_DIR}`, recipe variables from the `product_vars.yaml` file found in `${ADMIN_DIR}`, and `sat bootprep` input files found in `${ADMIN_DIR}`.
 
     (`ncn-m001#`) Execute the `update-cfs-config` and `prepare-images` stages.
 
     ```bash
-    iuf -a activity-all-products-upgrade -m "${ACTIVITY_DIR}" run --site-vars "${SITE_VARS_FILE}" --bootprep-config-dir "${ACTIVITY_DIR}/hpc-csm-software-recipe-23.05.0/vcs" -r update-cfs-config prepare-images
+    iuf -a "${ACTIVITY_NAME}" -m "${MEDIA_DIR}" run --site-vars "${ADMIN_DIR}/site_vars.yaml" --bpcd "${ADMIN_DIR}" -r update-cfs-config prepare-images
     ```
 
-    Once this step has completed:
+1. Inspect the newly-created management NCN and managed node images and CFS configurations to ensure they are correct before continuing with the next steps of the workflow.
+
+Once this step has completed:
+
+- New CFS configurations have been created for management NCNs and managed compute and application (UAN, etc.) nodes
+- New images have been created for management NCNs and managed compute and application (UAN, etc.) nodes
+- New BOS session templates have been created to boot managed compute and application (UAN, etc.) nodes with the new images and CFS configurations
+- Per-stage product hooks have executed for the `update-cfs-config` and `prepare-images` stages
+
+## Backup
+
+This section describes procedures that backup critical state in case it becomes necessary to fall back to previous configurations and software.
+
+TBD
```

Comment:
This is really only an upgrade step.   There won't be anything to back up for a fresh install.

Reply:
Fixed in next update.